design-documents/graph-ql/coverage/negotiableQuotes.graphqls

```diff
@@ -1,205 +1,332 @@
-    created_by: String @doc(description: "Negotiable quote creator")
-    created_at: String @doc(description: "Timestamp indicating when the negotiable quote was created.")
-    updated_at: String @doc(description: "Timestamp indicating when the negotiable quote was updated.")
-    payment_info: NegotiableQuotePaymentInfo! @doc(description: "Negotiable quote payment info.")
-    status: String @doc(description: "Negotiable quote status")
-    available_shipping_methods: [ShippingMethodsOutput]! @doc(description: "Available shipping methods")
+type CloseNegotiableQuotesOutput {
+    closed_quotes: [NegotiableQuote]! @doc(description: "Quotes that were just closed")
+    negotiable_quotes(
+        filter: NegotiableQuoteFilterInput,
+        pageSize: Int = 20,
+        currentPage: Int = 1
+    ): [NegotiableQuote] @doc(description: A (optionally filtered) list of Negotiable Quotes viewable by the logged-in customer")
 }
 
-input NegotiableQuoteFilterInput {
-    ids: FilterEqualTypeInput @doc(description: "Filter Customer Negotiable quotes with an negotiable quote ID or list of negotiable quote IDs")
-    name: FilterMatchTypeInput @doc(description: "Filter by display name of the negotiable quote")
+input RemoveNegotiableQuoteItemsInput {
+    quote_ID: ID!
+    quote_item_ids: [ID!]!
 }
 
-type NegotiableQuoteItemInterface {
-    id: ID!
-    quantity: Float!
-    stock: Float!
-    prices: NegotiableQuoteItemPrices
-    product: ProductInterface!
+input UpdateNegotiableQuoteQuantitiesInput {
+    quote_id: ID!
+    items: [NegotiableQuoteItemQuantityInput!]!
 }
 
-type NegotiableQuoteItemPrices {
-    price: Money!
-    row_total: Money!
-    row_total_including_tax: Money!
-    discounts: [Discount] @doc(description:"An array of discounts to be applied to the quote item")
-    total_item_discount: Money @doc(description:"The total of all discounts applied to the item")
+input NegotiableQuoteItemQuantityInput {
+    quote_item_id: ID!
+    quantity: Float!
 }
 
-type Discount @doc(description:"Defines an individual discount. A discount can be applied to the quote as a whole or to an item.") {
-    amount: Money! @doc(description:"The amount of the discount")
-    label: String! @doc(description:"A description of the discount")
+input RequestNegotiableQuoteInput {
+    cart_id: ID!
+    quote_name: String!
+    comment: NegotiableQuoteCommentInput!
+    # files (attachments) to be added at a later date when file upload design has been finalized
 }
 
-type DefaultNegotiableQuoteItem implements NegotiableQuoteItemInterface
-@doc(description: "Negotiable Quote Item Implementation for Simple and Virtual Products") {
-    id: ID! @doc(description: "Unique Identifier of Negotiable Quote Item")
-    product: ProductInterface!
-    qty: Float! @doc(description:  "Quantity added")
-    customizable_options: [SelectedCustomizableOption] @doc(description: "custom Option selected")
+input NegotiableQuoteCommentInput {
+    comment: String!
```

Comment:
Quote ID seems to be missing.

Reply:
This input is used in multiple places, for multiple scenarios.

**Scenarios**
- Add a comment/attachment while creating a negotiable quote (`Mutation.requestNegotiableQuote`)
- Add a comment/attachment to an existing quote (`Mutation.addNegotiableQuoteComment`)

The same `NegotiableQuoteCommentInput` input type is used for both these cases. For the second scenario, the Quote ID is in `AddNegotiableQuoteCommentInput`